common/chunkStatusLogger.go

```diff
@@ -337,13 +347,19 @@ func (csl *chunkStatusLogger) main(chunkLogPath string) {
 			alwaysFlushFromNowOn = true
 			doFlush()
 			csl.flushDone <- struct{}{}
-			continue // TODO can become break (or be moved to later if we close unsaved entries, once we figure out how we got stuff written to us after CloseLog was called)
+			select {
+			case <-csl.done:
+				return
```

Comment:
again, a comment here will be useful

// If we came here because CloseLogger() was called, there won't be any more chunks to log, end the processing loop.